clients/drcachesim/tests/burst_longblock.cpp

```diff
@@ -0,0 +1,323 @@
+                             "fpatan\n\t" // 160
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t" // 170
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t" // 180
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t" // 190
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t"
+                             "fpatan\n\t" // 200
+                             "mov (%rsp),%rax\n\t");
+#else
+        // It is not worth the effort to replicate on other platforms.
+#    error Not implemented
+#endif
+    }
+    return nullptr;
+}
+
+static int
+do_some_work()
+{
+    // To reduce the trace size we just sleep and let the other thread
+    // be the only one generating data.
+    constexpr int MILLIS_TO_SLEEP = 10;
+    std::this_thread::sleep_for(std::chrono::milliseconds(MILLIS_TO_SLEEP));
+    return 1;
+}
+
+static void
+gather_trace()
+{
+    // We need -no_align_endpoints to make reproducing the bug much more likely.
+    // Otherwise, the tracer switches to nop mode and the detach happens mid-block
+    // but tracing has already ended.
+    if (!my_setenv("DYNAMORIO_OPTIONS",
+                   "-stderr_mask 0xc -client_lib ';;-offline -no_align_endpoints"))
+        std::cerr << "failed to set env var!\n";
+    std::cerr << "pre-DR init\n" << std::flush;
+    dr_app_setup_and_start();
+    assert(dr_app_running_under_dynamorio());
+    if (do_some_work() < 0)
+        std::cerr << "error in computation\n";
+    // TODO i#6490: This app produces incorrect output when run under DR if we do
```

Comment:
Sigh. I thought we fixed this.

Reply:
Oh, I didn't observe this: I admit I copy-pasted parts of this from another test. The comment is also in burst_gencode and burst_syscall_inject. So presumably these can all be removed: removing from longblock here; will send a separate PR to remove from the others.

Reply:
I removed here and the test does pass 10 times in a row.

Reply:
To clarify, I removed both the comment and all the "std::flush" calls.